File: HealthSysHub.Web.Utility/Models/PharmacyPaymentDetail.cs
namespace HealthSysHub.Web.Utility.Models
{
    public class PharmacyPaymentDetail : PharmacyOrderDetails
    {
        public Guid? PaymentId { get; set; }
        public Guid? HospitalId { get; set; }
        public string? PaymentNumber { get; set; }
        public DateTimeOffset? PaymentDate { get; set; }
        public string? PaymentMethod { get; set; }
        public decimal? PaymentAmount { get; set; } 
        public string? ReferenceNumber { get; set; }
        public string? PaymentStatus { get; set; }
        public string? PaymentGateway { get; set; }
        public string? GatewayResponse { get; set; }
        public string? PaymentNotes { get; set; }
    }
}

File: HealthSysHub.Web.Pharmacy.UI/src/app/models/pharmacy-payment-detail.ts
import { PharmacyOrderDetails } from "./pharmacy-order-details";

export interface PharmacyPaymentDetail extends PharmacyOrderDetails {
  paymentId?: string; // Guid as string
  hospitalId?: string; // Guid as string
  paymentNumber?: string | null;
  paymentDate?: string | null; // ISO string for DateTimeOffset
  paymentMethod?: string | null;
  paymentAmount?: number | null;
  referenceNumber?: string | null;
  paymentStatus?: string | null;
  paymentGateway?: string | null;
  gatewayResponse?: string | null;
  paymentNotes?: string | null;
}